lib/lf/geometry/test/geometry_tests.cc

```diff
@@ -98,6 +98,43 @@ void checkIntegrationElement(const lf::geometry::Geometry &geom,
   }
 }
 
+/**
+ * Checks that SubGeometry and Geometry map the same nodes to the same points
+ */
+void checkSubGeometry(const lf::geometry::Geometry &geom) {
+  // nodeCoords is a (refEl.Dimension, refEl.NumNodes) matrix
+  const auto refEl = geom.RefEl();
+  const Eigen::MatrixXd &nodeCoords = refEl.NodeCoords();
+
+  // iterate over all relative codimensions
+  for (size_t codim = 1; codim <= geom.RefEl().Dimension(); ++codim) {
```

Comment:
special case for codim=0: In this case the geometry object should be "cloned", i.e. it should return a new geometry object that represents exactly the same mapping. I think you can just extend the loop to codim=0 and it should test this behavior.

Reply:
Yes, thanks